src/parser.ts

```diff
@@ -222,30 +224,15 @@ const parseRouteEndpoint = async (route: IRoute, basePath: string, config: IGene
 				}
 			})
 
-			const [workflowHandler, ...permissionHandlers] = await Promise.all([workflowHandlerPromise as Promise<ILocation> | null, ...permissionHandlerPromises])
-
-			// handle permissions
-			const permissions: { [groupName: string]: string[] } = {}
+			const [workflowHandler, ...permissionHandlers] = await Promise.all([workflowHandlerPromise as Promise<ILocation> | null, ...middlewaresHandlerPromises])
 
+			// handle middleware attributeValues
+			const middlewares: IEndpointMiddleware[] = []
 			forEach(permissionHandlers, (permissionHandler) => {
-				if (permissionHandler.groupName) {
-					if (!permissions[permissionHandler.groupName]) {
-						permissions[permissionHandler.groupName] = []
-					}
-				} else if (!permissions.default) {
-					permissions.default = []
-				}
-
-				forEach(permissionHandler.resultProperties.result, (value) => {
-					if (!isNaN(parseInt(value.name, 10))) {
-						const permission = value.value.value
-
-						if (permissionHandler.groupName) {
-							permissions[permissionHandler.groupName].push(permission)
-						} else {
-							permissions.default.push(permission)
-						}
-					}
+				middlewares.push({
+					closure: permissionHandler.closure,
```

Comment:
@kot-1999 for what do you need to return closure here? where is it used after this?

Reply:
Actually I don't